applications/TopologyOptimizationApplication/custom_elements/comparison_utilities.h

```diff
@@ -0,0 +1,342 @@
+//
+//   Project Name:        KratosSolidMechanicsApplication $
+//   Created by:          $Author:            JMCarbonell $
+//   Last modified by:    $Co-Author:                     $
+//   Date:                $Date:                July 2013 $
+//   Revision:            $Revision:                  0.0 $
+//
+//
+
+#if !defined(KRATOS_COMPARISON_UTILITIES )
+#define  KRATOS_COMPARISON_UTILITIES
+
+
+/* System includes */
+
+/* Project includes */
+#include "math_utilities.h"
+
+
+namespace Kratos
+{
+
+/**@name Kratos Globals */
+/*@{ */
+
+
+/*@} */
+/**@name Type Definitions */
+/*@{ */
+
+/*@} */
+
+
+/**@name  Enum's */
+/*@{ */
+
+
+/*@} */
+/**@name  Functions */
+/*@{ */
+
+
+
+/*@} */
+/**@name Kratos Classes */
+/*@{ */
+
+
+class ComparisonUtilities
+{
+public:
+    /**@name Type Definitions */
+    /*@{ */
+
+
+    /*@} */
+    /**@name Life Cycle
+     */
+    /*@{ */
+
+    /** Constructor.
+     */
+    ComparisonUtilities() {};
+
+    /** Destructor.
+     */
+    ~ComparisonUtilities() {};
+
+    /** Operators.
+     */
+
+    //**************************************************************************
+    //**************************************************************************
+
+
+    //**************************************************************************
+    //**************************************************************************
+
+    double CalculateStressNorm(const Vector& StressVector)
+    {
+        KRATOS_TRY
+
+        Matrix LocalStressTensor  = MathUtils<double>::StressVectorToTensor(StressVector); //reduced dimension stress tensor
+
+	Matrix StressTensor(3,3); //3D stress tensor
+	noalias(StressTensor) = ZeroMatrix(3,3);
+	for(unsigned int i=0; i<LocalStressTensor.size1(); i++)
+	  {
+	    for(unsigned int j=0; j<LocalStressTensor.size2(); j++)
+	      {
+		StressTensor(i,j) = LocalStressTensor(i,j);
+	      }
+	  }
+
+        double StressNorm =  ((StressTensor(0,0)*StressTensor(0,0))+(StressTensor(1,1)*StressTensor(1,1))+(StressTensor(2,2)*StressTensor(2,2))+
+			      (StressTensor(0,1)*StressTensor(0,1))+(StressTensor(0,2)*StressTensor(0,2))+(StressTensor(1,2)*StressTensor(1,2))+
+			      (StressTensor(1,0)*StressTensor(1,0))+(StressTensor(2,0)*StressTensor(2,0))+(StressTensor(2,1)*StressTensor(2,1)));
+
+	StressNorm = sqrt(StressNorm);
+
+        return StressNorm;
+
+        KRATOS_CATCH( "" )
+
+    };
+
+    //**************************************************************************
+    //**************************************************************************
+
+    double CalculateVonMises(const Vector& StressVector)
```

Comment:
VM is already calculated in the utilities of the StructuralMechanicsApplication, please use that

Reply:
THe function `CalculateVonMisesEquivalentStress`?

Reply:
Yes

Reply:
What do I use as `VoigtSize` in this case?

Reply:
In 3D is 6, in 2D is 3 (I recommend to see the tests and current implementations as references)